cls/SourceControl/Git/Utils.cls

```diff
@@ -1533,6 +1630,29 @@ ClassMethod RunGitCommandWithInput(command As %String, inFile As %String = "", O
     quit returnCode
 }
 
+ClassMethod GenerateCommitMessageFromFiles(filesWithActions) As %String
+{
+    set commitMsg = ""
+
+    set iterator = filesWithActions.%GetIterator()
+    while iterator.%GetNext(,.value,) {
+        set action = value.%Get("action")
+        set file = value.%Get("file")
+
+
+        if action = "A" {
```

Comment:
Perhaps restructure as:

```
set oneFileMsg = $Case(action,
  "A":"added",
  "M":"modified",
  "D":"deleted",
  "R":"renamed",
  :"???")_" "_file
```

It's a bit less code and will also handle (to some extent) other operations, if they happen to exist - do you have an exhaustive list of codes?